tools/validate-manifest/src/Manifest.ts

```diff
@@ -0,0 +1,200 @@
+import Ajv, { JSONSchemaType } from "ajv";
+
+/**
+  A Manifest maps template IDs to template app specifications.
+ */
+export type Manifest = { [id: string]: TemplateSpec };
+
+/**
+  A template app is a backend configuration plus (optionally) some frontend
+  client source code.
+ */
+export type TemplateSpec = BackendOnlyTemplateSpec &
+  Partial<ClientTemplateSpec>;
+
+/**
+  A template app can be just a backend configuration without a client.
+ */
+export type BackendOnlyTemplateSpec = {
+  /**
+    The friendly name of the template app.
+   */
+  name: string;
+
+  /**
+    Obsolete. The name of the GitHub org or user with the template source.
+   */
+  repo_owner: string;
+
+  /**
+    Obsolete. The name of the GitHub repo where the template source can be found.
+   */
+  repo_name: string;
+
+  /**
+    The path to the exported App Services backend configuration directory.
+   */
+  backend_path: string;
+};
+
+/**
+  A template app can have a client.
+ */
+export type ClientTemplateSpec = BackendOnlyTemplateSpec & {
```

Comment:
nit: would it make sense to rename this here since it's a backend _and_ client template spec?

Reply:
My thinking was that the more important thing is that it defines a client and merely refers to a backend. I had `WithClientTemplateSpec` at one point but it seemed excessive.